test/SILOptimizer/mem2reg_lifetime.sil

```diff
@@ -1858,8 +1858,7 @@ right:
 // CHECK:       {{bb[^,]+}}([[INSTANCE:%[^,]+]] : @owned $C):
 // CHECK:         cond_br undef, [[BB1:bb[0-9]+]], [[BB4:bb[0-9]+]]
 // CHECK:       [[BB1]]:
-// CHECK:         [[COPY:%[^,]+]] = copy_value [[INSTANCE]]
```

Comment:
I'm curious why this change eliminated a copy in mem2reg.

Reply:
This copy was incorrectly introduced in https://github.com/apple/swift/pull/68608/files#diff-8bd51ca385c836620fb33d8b5480d0ab38024c0ba38f83aa599f0a5148c48e7fR1861 .  It was introduced because the unreachable instruction was added to liveness although the value is not available at it.  This is fixed here by not adding unavailable terminators to liveness.